IHP/RouterSupport.hs

```diff
@@ -68,9 +69,81 @@ import qualified IHP.WebSocket as WS
 import GHC.TypeLits as T
 import IHP.Controller.Context
 import IHP.Controller.Param
+import qualified Data.TMap as TMap
+import qualified IHP.ApplicationContext as ApplicationContext
+
+putContextRouter :: forall value. (Typeable value) => value -> RouteParser -> RouteParser
```

Comment:
Why can't we use the existing controller context also during routing?

Right now the controller context is created when the action is called. But it seems there's nothing preventing us from creating the controller context before routing.

Then we could just use `putContext` eveywhere, which would simplify the public API of IHP.

Reply:
`putContext` is an `IO` function. So parsing would require constant lifts, which to me seemed less nice than having to use a special `putContextRouter` function for parsing.

Reply:
Ok this makes sense 👍